src/app/detail/ui/Sidebar.tsx

```diff
@@ -15,61 +15,74 @@ import { UpdateTaskRequest } from '@/types/dto/tasks.dto'
-import { NoAssignee, NoAssigneeExtraOptions } from '@/utils/noAssignee'
-import ExtraOptionRendererAssignee from '@/components/inputs/ExtraOptionRendererAssignee'
+import { NoAssignee } from '@/utils/noAssignee'
 import { WorkflowStateSelector } from '@/components/inputs/Selector-WorkflowState'
 import { CopilotAvatar } from '@/components/atoms/CopilotAvatar'
 import { AssigneePlaceholder } from '@/icons'
-import { useState } from 'react'
-import { MiniLoader } from '@/components/atoms/MiniLoader'
+import { useEffect, useState } from 'react'
 import { setDebouncedFilteredAssignees } from '@/utils/users'
 import { z } from 'zod'
 import { isAssigneeTextMatching } from '@/utils/assignee'
-import { DateStringSchema } from '@/types/date'
 
 const StyledText = styled(Typography)(({ theme }) => ({
   color: theme.color.gray[500],
   width: '80px',
 }))
 
 export const Sidebar = ({
+  task_id,
   selectedWorkflowState,
   selectedAssigneeId,
   updateWorkflowState,
-  dueDate,
   updateAssignee,
   updateTask,
   assignee,
   disabled,
 }: {
+  task_id: string
   selectedWorkflowState: WorkflowStateResponse
   selectedAssigneeId: string | undefined
-  dueDate: string | undefined
   updateWorkflowState: (workflowState: WorkflowStateResponse) => void
   updateAssignee: (assigneeType: string | null, assigneeId: string | null) => void
   updateTask: (payload: UpdateTaskRequest) => void
   assignee: IAssigneeCombined[]
   disabled: boolean
 }) => {
-  const { workflowStates, token } = useSelector(selectTaskBoard)
+  const { tasks, token, workflowStates } = useSelector(selectTaskBoard)
   const { showSidebar } = useSelector(selectTaskDetails)
   const [filteredAssignees, setFilteredAssignees] = useState(assignee)
   const [activeDebounceTimeoutId, setActiveDebounceTimeoutId] = useState<NodeJS.Timeout | null>(null)
   const [loading, setLoading] = useState(false)
+  const [dueDate, setDueDate] = useState<Date | string | undefined>()
 
   const { renderingItem: _statusValue, updateRenderingItem: updateStatusValue } = useHandleSelectorComponent({
-    item: selectedWorkflowState,
+    // item: selectedWorkflowState,
+    item: null,
     type: SelectorType.STATUS_SELECTOR,
   })
   const { renderingItem: _assigneeValue, updateRenderingItem: updateAssigneeValue } = useHandleSelectorComponent({
-    item: selectedAssigneeId ? assignee.find((el) => el.id === selectedAssigneeId) : NoAssignee,
+    // item: selectedAssigneeId ? assignee.find((el) => el.id === selectedAssigneeId) : NoAssignee,
+    item: null,
     type: SelectorType.ASSIGNEE_SELECTOR,
   })
 
   const statusValue = _statusValue as WorkflowStateResponse //typecasting
   const assigneeValue = _assigneeValue as IAssigneeCombined //typecasting
 
+  useEffect(() => {
+    if (tasks && workflowStates) {
+      const currentTask = tasks.find((el) => el.id === task_id)
+      const currentWorkflowState = workflowStates.find((el) => el?.id === currentTask?.workflowStateId)
```

Comment:
This should be unnecessary since we are including `workflowState` in the tasks API response right? Unless you are doing some filtering to remove the workflowState key, after you find the current task the `workflowState` key will contain the necessary info and another `.find` should be unnecessary.

Reply:
This is necessary here because the tasks response doesn't provide the complete information of the workflowState which are required here.